README.md

```diff
@@ -1 +1,95 @@
+    |                                                         |
+    | Please select the location of source code:              |
+    |  * WD342112TX 500GB                                     |
+    |    - p1 100GB ext3   <-                                 |
+    |    - p2 30GB fat                                        |
+    |  * ADATA 8GB                                            |
+    |                                                         |
+    +---------------------------------------------------------+
+
+Then, the system must open a menu to user select the file or create a
+new one on a specified directory. 
+
+Example file selection from MikeOS:
+
+![select file](http://i.stack.imgur.com/mr8H2.png)
+
+After that, the assembler IDE will
+show up.
+
+    +---------------------------------------------------------+
+    | Save (C-s) | Open (C-o) | Assemble (C-c) | Test (C-x)   |
+    +---------------------------------------------------------+
+    |  Install   | Show info (C-h)| Reset (C-r)| Shutdown     |
+    +---------------------------------------------------------+
+    |        ;;; Example operating system                     |
+    |start:                                                   |
+    |        mov ah, 2     ;; set cursor position             |
+    |        mov dh, 3     ;; move to row 3                   |
+    |        mov dl, 14    ;; column 14                       |
+    |        int 10h       ;; BIOS Video service call         |
+    |                                                         |
+    |        dd 0xdeadbeef ;; invalid opcode will return      |
+    |                      ;; the control to the IDE          |
+    |                                                         |
+    |                                                         |
+    |                                                         |
+    |                                                         |
+    |                                                         |
+    +---------------------------------------------------------+
+    | Assembler output:                                       |
+    | Built successfully at memory address 0x1000-0x11ff      |
+    +---------------------------------------------------------+
+    | Processor: Intel(R) Core(TM) i3-3217U (Real mode)       |
+    | Memory: 4GB  Used: 300KB                                |
+    +---------------------------------------------------------+
+
+When pressed C-c the system must assemble the file in-memory and when
+pressed C-x the system must prepare the processor and jump to
+assembled location as if it had been executed by BIOS (no reboot).
+
+The system must have the IVT (Interrupt Vector Table) configured to
+handle invalid opcode exceptions.
+
+The user developed operating system could take full control of the
+machine, remapping IVT, enter protected mode, set GDT, ISR, etc, and
+reuse EnzOS memory. But in this case, it'll require a reboot to get
+into IDE again.
```

Comment:
That depends. 

The example code loaded in the editor have an invalid instruction in the end, that will throw a processor exception that we can handle and restore the IDE state. But this requires that the user-assembled code do not remap this...

If the user OS runs in protected mode and remap everything, then (I think) the only way to get the control back to the IDE is using some kind of virtualization (hypervisor, etc).

I don't think a reboot is something bad in this case, because we can save the IDE memory state on disk and load at startup very fast.

But we can think in other ways to improve this. Maybe in the future we can use hypervisor features to achieve this in the best way.